jina/peapods/runtimes/zmq/zed.py

```diff
@@ -204,26 +204,42 @@ def _handle(self) -> 'ZEDRuntime':
             Handle does not handle explicitly message because it may wait for different messages when different parts are expected
         :return: ZEDRuntime procedure.
         """
-
-        if (
-            not re.match(self.envelope.header.target_peapod, self.name)
-            or self.request_type != 'DataRequest'
-        ):
+        # skip executor for non-DataRequest
+        if self.request_type != 'DataRequest':
+            self.logger.debug(f'skip executor: not data request')
             return self
 
         # migrated from the previously RouteDriver logic
+        # set dealer id
         if self._idle_dealer_ids:
             dealer_id = self._idle_dealer_ids.pop()
             self.envelope.receiver_id = dealer_id
 
             # when no available dealer, pause the pollin from upstream
             if not self._idle_dealer_ids:
                 self._zmqstreamlet.pause_pollin()
+            self.logger.debug(
+                f'using route, set receiver_id: {self.envelope.receiver_id}'
+            )
 
+        # skip executor if target_peapod mismatch
+        if (
+            self.envelope.header.target_peapod
+            and self.envelope.header.target_peapod != self.name.split('/')[0]
```

Comment:
behavior change without reason and explain to user. please revert back to `re.match(self.envelope.header.target_peapod, self.name)`

Reply:
After checking the [docstring](https://github.com/jina-ai/jina/blob/master/.github/2.0/cookbooks/Flow.md#function-signature), I see your point. However, I need to make it clear there will be an unexpected behavior, `target_peapod='foo'` will match both pods with the name `foo` and `foo_what_ever_follows`

Reply:
I've added a warning message in the docstring at jina/clients/mixin.py